source/includes/config-repo/_022-pipeline-export.md.erb

```diff
@@ -0,0 +1,132 @@
+
+> The plugin will receive the following JSON body —
+
+```json
+{
+  "pipeline": {
+    "group": "my-group",
+    "name": "my-pipeline",
+    "label_template": "${COUNT}",
+    "lock_behavior": "none",
+    "mingle": {
+      "mql_grouping_conditions": "\"\""
+    },
+    "environment_variables": [],
+    "parameters": [],
+    "materials": [
+      {
+        "url": "https://my.repo/path/to.git",
+        "branch": "master",
+        "shallow_clone": false,
+        "filter": {
+          "ignore": [],
+          "whitelist": []
+        },
+        "auto_update": true,
+        "type": "git"
+      }
+    ],
+    "stages": [
+      {
+        "name": "first-stage",
+        "fetch_materials": true,
+        "never_cleanup_artifacts": false,
+        "clean_working_directory": false,
+        "approval": {
+          "type": "success",
+          "users": [],
+          "roles": []
+        },
+        "environment_variables": [],
+        "jobs": [
+          {
+            "name": "first-job",
+            "environment_variables": [],
+            "tabs": [],
+            "resources": [],
+            "artifacts": [],
+            "properties": [],
+            "run_instance_count": "0",
+            "tasks": [
+              {
+                "command": "/bin/echo",
+                "timeout": -1.0,
+                "arguments": [
+                  "hello"
+                ],
+                "run_if": "passed",
+                "type": "exec"
+              }
+            ]
+          }
+        ]
+      }
+    ]
+  }
+}
+```
+
+<h3><!-- Force a break/clear --></h3>
+
+<p class='response-code-heading'>Response code</p>
+
+The plugin is expected to return status `200` if it can understand the request.
+
+<p class='response-body-heading'>Response body</p>
+
+The plugin is expected to return a JSON object (corresponding to
+[`PipelineExportResponseMessage`](https://github.com/gocd/gocd/blob/b8cbce628afca8b20726e5fbd81ebea1e20091fe/plugin-infra/go-plugin-access/src/main/java/com/thoughtworks/go/plugin/access/configrepo/v2/messages/PipelineExportResponseMessage.java). The response body contains a single attribute:
+
+#### `pipeline`
+
+This `pipeline` element is a string representation of the exported pipeline configuration in the native language supported by the plugin. The value of this element must deserialize to a `string`.
+
+<p class="response-headers-heading">Response headers</p>
+
+The plugin response must also set the following headers:
+
+#### `Content-Type`
+
+Plugin responses are expected to provide a MIME type to indicate how to treat the content of the `pipeline` attribute. Here are some examples of common MIME types:
+
+* `text/plain; charset=utf-8`
+* `application/json; charset=utf-8`
+* `application/x-yaml; charset=utf-8`
+
+#### `X-Export-Filename`
+
+Plugin responses must also suggest a filename for the exported content. This is used by GoCD to construct a `Content-Disposition` header for REST API responses.
```

Comment:
Can you provide an example, similar to any API response? Say: https://api.gocd.org/current/#get-all-artifacts

With the expected headers shown and a random YAML/JSON body too.

Reply:
Ok.